### PR TITLE
adapt MiniCPM-Llama3-V-2_5 new code

### PR DESCRIPTION
**Pull request by irexyc**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily receiving feedbacks. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

Please describe the motivation of this PR and the goal you want to achieve through this PR.

## Modification

Please briefly describe what modification is made in this PR.

## BC-breaking (Optional)

Does the modification introduce changes that break the backward-compatibility of the downstream repositories?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here, and update the documentation.

## Checklist

1. Pre-commit or other linting tools are used to fix the potential lint issues.
2. The modification is covered by complete unit tests. If not, please add more unit tests to ensure the correctness.
3. If the modification has a dependency on downstream projects of a newer version, this PR should be tested with all supported versions of downstream projects.
4. The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

File: lmdeploy/vl/model/minicpmv.py [hunk 1]
```diff
@@ -54,6 +54,20 @@ def build_model(self):
...
 
+        # adapt new code commit 287e3f85
+        if not hasattr(model, 'slice_image'):
+            from transformers import AutoProcessor
+            processor = AutoProcessor.from_pretrained(self.model_path,
+                                                      trust_remote_code=True)
+            model.slice_image = processor.image_processor.slice_image
+
+            def _reshape_by_patch(x):
+                x = x.numpy()
```
Comment (RunningLeon) [inline lmdeploy/vl/model/minicpmv.py:65]: if x is on the device, should it be `x.cpu().numpy()`?

File: lmdeploy/vl/model/minicpmv.py [hunk 2]
```diff
@@ -54,6 +54,20 @@ def build_model(self):
...
+                x = processor.image_processor.reshape_by_patch(x)
+                return torch.from_numpy(x)
```
Comment (RunningLeon) [inline lmdeploy/vl/model/minicpmv.py:67]: should we set the device of original input x.device?

File: lmdeploy/vl/model/minicpmv.py (hunk 1 above)
Comment (irexyc) [reply lmdeploy/vl/model/minicpmv.py:65]: I think we don't need to make it too complicated and follow the current upstream implemention is enough.

Currently, x(slice_image) is cpu tensor. 
https://huggingface.co/openbmb/MiniCPM-Llama3-V-2_5/blob/main/modeling_minicpmv.py#L53-L61

Comment (RunningLeon) [reply lmdeploy/vl/model/minicpmv.py:65]: Personally, it's safe to do so.

Comment (lvhan028) [reply lmdeploy/vl/model/minicpmv.py:65]: I agree with @RunningLeon

Using `x.cpu().numpy()` is not too complicated. Moreover, it is safe

**Review by RunningLeon — APPROVED** [review]
LGTM

**Review by lvhan028 — APPROVED** [review]

**Pull request merged**